### PR TITLE
Refs #12635 - added migration for grub tftp dirs

### PR DESCRIPTION
**Pull request by lzap**
Required change for https://github.com/theforeman/puppet-foreman_proxy/pull/276

### CODE REVIEW

**Comment by mmoll** [comment]
this seems to fail

File: config/foreman.migrations/20160831180000_foreman_proxy_tftp_grub2.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+root = answers['foreman_proxy']['tftp_root']
```
Comment (domcleal) [inline config/foreman.migrations/20160831180000_foreman_proxy_tftp_grub2.rb]: `answers['foreman_proxy']` may be false if the module is disabled

File: config/foreman.migrations/20160831180000_foreman_proxy_tftp_grub2.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+dirs = answers['foreman_proxy']['tftp_dirs']
+dirs << "#{root}/grub"
```
Comment (domcleal) [inline config/foreman.migrations/20160831180000_foreman_proxy_tftp_grub2.rb]: `dirs` may be nil on the first run, not necessarily an Array

**Comment by lzap** [comment]
Damn, I am unable to get to my work tree this week. Upgrades will be likely broken for RC1 then, will fix this for RC2 as my top priority.

File: config/foreman.migrations/20160831180000_foreman_proxy_tftp_grub2.rb (hunk 2 above)
Comment (lzap) [reply config/foreman.migrations/20160831180000_foreman_proxy_tftp_grub2.rb]: Nil on the first run? Why is that?

**Comment by lzap** [comment]
I made a change using Github editor. @mbacovsky we talked about this in Brno last week, is this correct?

File: config/foreman.migrations/20160831180000_foreman_proxy_tftp_grub2.rb (hunk 2 above)
Comment (domcleal) [reply config/foreman.migrations/20160831180000_foreman_proxy_tftp_grub2.rb]: The first run will be using the default answers file which doesn't contain much: https://github.com/theforeman/foreman-installer/blob/develop/config/foreman-answers.yaml#L14

**Comment by domcleal** [comment]
Merged, thanks @lzap.

**Pull request merged**